File: two_pointers/two_sum.py

target = 10

array = [1,2,3,4,5,6,7,8,9,10,11]

low = 0


count = len(array)-1

high = count

answer = []

i = high

first = 0
second = 0

#find the larger number
while i > 0:
    if array[i] >= target:
      i-=1
    else:
      first = array[i]
      answer.append( array[i])
      break

#find the smaller number
i = 0
gap = target - first

while i < len(array):
    if array[i] != gap:
        i+=1
    else:
        second = array[i]
        answer.append( array[i])
        break

print(answer)

File: two_pointers/two_sum.c
#include <stdio.h>

int doubleSum(int array[], int target, int count){
  int first = 0;
  int second = 0;

  int i = count - 1;

  while (i >= 0) {
    
    if (array[i] >= target){
      i--;
    }
    else{
      first = array[i];
    break;
    }

    
    
  }

 printf("The first number is %d\n", first);

 int gap = target - first;
 i = 0;

 while (i <= count) {

   if (array[i] != gap) {
     
     i++;
   } else {
     second = array[i];
     break;
   }

 }
 
 printf("The second number is %d", second);
  return 0;
}

int main(void) {

  int array[] = {1,2,3,4,5,6,7,8,9,10,11};

  int target = 5;

  int count = sizeof(array)/sizeof(array[0]);

  doubleSum(array, target, count);

  return 0;
}
